src/plug/bash_plugin.rs

```diff
@@ -1,40 +1,79 @@
-use std::process::Command;
-use std::path::{Path,PathBuf};
+use crate::plug::{unwrap_home_path, CallablePlugin, PluginInterface, PLUGINS_PATH};
+use crate::plugin_logger::PluginLogger;
 use std::collections::HashMap;
-use crate::plug::{unwrap_home_path, PluginInterface, CallablePlugin, PLUGINS_PATH};
-
+use std::fmt::Display;
+use std::io::BufReader;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
 
 struct BashCommand {
     executable: PathBuf,
 }
 
-
 impl PluginInterface for BashCommand {
-    fn run(&self, config: &HashMap<String, String>) -> Result<(), String> {
+    fn run(
+        &self,
+        name: &str,
+        config: &HashMap<String, String>,
+    ) -> Result<Box<dyn Display>, String> {
         // get a string of args and values from hashmap
-        let output = Command::new("bash")
-             .envs(config)
-             .arg("-C")
-             .arg(self.executable.to_str().unwrap())
-             // .args(&args)
-             .output()
-             .expect("failed to execute process {}");
-
-        println!("{}", String::from_utf8(output.stdout).unwrap());
-        Ok(())
+        let mut child = Command::new("bash")
+            .envs(config)
+            .arg("-C")
+            .arg(self.executable.to_str().unwrap())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()
+            .expect("failed to start process {}");
+
+        // Wrap streams into a buffered stream, then box it
+        let stdout = Box::new(BufReader::new(
+            child.stdout.take().expect("Stdout is unavailable"),
+        ));
+        let stderr = Box::new(BufReader::new(
+            child.stderr.take().expect("Stderr is unavailable"),
+        ));
+
+        // Init logger
+        let plugin_logger = PluginLogger::new(name, vec![stdout, stderr]);
```

Comment:
Feels like the whole logging logic should be a separate trait for a plugin, since so far handling everything inside the plugin interface violates SOLID so badly. 

One of the solutions I see is to have a stdout/stderr pipe on plugin interface that we can suck on and get all the output from. Then some Logger trait can use such pipes to format and redirect to the kitty's stdout or to the file.